src/portal-impl.c

```diff
@@ -29,6 +29,35 @@
 #include <glib.h>
 #include <gio/gio.h>
 
+typedef struct _PortalInterface {
+  /* dbus_name is NULL if this is the default */
+  char *dbus_name;
+  char **portals;
+} PortalInterface;
+
+typedef struct _PortalConfig {
+  char *source;
+  PortalInterface **ifaces;
+  size_t n_ifaces;
+} PortalConfig;
+
+static void
+portal_interface_free (PortalInterface *iface)
+{
+  g_free (iface->dbus_name);
+  g_strfreev (iface->portals);
+  g_free (iface);
+}
+
+static void
+portal_config_free (PortalConfig *config)
+{
+  g_free (config->source);
+  for (size_t i = 0; i < config->n_ifaces; i++)
+    portal_interface_free (config->ifaces[i]);
+  g_free (config);
```

Comment:
`g_free()` is NULL-safe, guaranteed by the C standard, and clearing the pointer right before destruction is just going to be extra work.

Anyway, I can change the rest of the destructors to do that.

Reply:
Sure, I'm familiar with these guarantees, I've asked this not because of NULL-safety but rather to keep things a bit more consistent from now on :slightly_smiling_face: